File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/common/viewmodels/converters/ApiListMovieConverter.java
package com.crpr.androidcinema.domain.common.viewmodels.converters;

import com.crpr.androidcinema.data.api.models.ApiMovie;
import com.crpr.androidcinema.data.api.models.enums.Size;
import com.crpr.androidcinema.domain.common.providers.ImageUrlProvider;
import com.crpr.androidcinema.domain.common.viewmodels.Converter;
import com.crpr.androidcinema.domain.discover.ListMovieModel;

/**
 * Created by claudioribeiro on 13/07/16.
 */
public class ApiListMovieConverter implements Converter<ListMovieModel, ApiMovie> {

    @Override
    public ListMovieModel map(ApiMovie parseObject) {
        String url = ImageUrlProvider.sharedInstance().getUrlFor(Size.W300);

        if(url != null){
            url += parseObject.getBackdropPath();
        }

        return ListMovieModel.id(parseObject.getId())
                                .title(parseObject.getTitle())
                                .imagePath(url)
                                .popularity(parseObject.getPopularity())
                                .votesAvg(parseObject.getVoteAverage())
                                .build();
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/common/viewmodels/converters/ApiListMovieConverter.kt
package com.crpr.androidcinema.domain.common.viewmodels.converters

import com.crpr.androidcinema.data.api.models.ApiMovie
import com.crpr.androidcinema.data.api.models.enums.Size
import com.crpr.androidcinema.domain.common.providers.ImageUrlProvider
import com.crpr.androidcinema.domain.common.viewmodels.Converter
import com.crpr.androidcinema.domain.discover.ListMovieModel

/**
 * Created by claudioribeiro on 13/07/16.
 */
class ApiListMovieConverter : Converter<ListMovieModel, ApiMovie>{

    override fun mapTo(parseObject: ApiMovie): ListMovieModel {
        var url: String? = ImageUrlProvider.sharedInstance().getUrlFor(Size.W300)

        if (url != null) {
            url += parseObject.backdrop_path
        }

        return ListMovieModel.id(parseObject.id)
                .title(parseObject.title!!)
                .imagePath(url!!)
                .popularity(parseObject.popularity)
                .votesAvg(parseObject.vote_average)
                .build()
    }

}
